### ISSUE TITLE
Multi provider deployments

### ISSUE THREAD

**Issue by medikoo** [serverless/serverless#9583]
### Use case description

In scope of a single service we'd like to be able to configure resources which are deployed to multiple providers.

When planning also it'll be good to consider
- __Modular approach__: Derive from single `serverless` installation, instead switch to model where each construct which is in sope of interest is installed individually, and `serverless` just serves as a core CLI that orchestrates deployments and other CLI commands
- __Any and multi region deployments__: - So within one service we can deploy different constructs to different regions, and so given constructs can be deployed to multiple regions at one
- __Fast development__:  Ideally through great cloud integration, but do not close the door for _offline emulation_ (while it's  being promoted as an anti-pattern, users should be able to achieve that if they wish so)
- __Any provider deployments__: In basic cases, provide a mean to easily switch given _construct_ to be deployed to other provider

### Proposed solution

We should specify:
- End Configuration structure 
- End Internal Processing flow
- Roadmap of implementation changes which should get us to desired result

_TBA_

**Comment by leonardotrindade** [comment]
Hi,

This idea still alive? Looking forward to this

**Comment by medikoo** [comment]
@leonardotrindade Yes, it's alive, and is actually sort of already served by the [Serverless Compose](https://www.serverless.com/framework/docs/guides/compose), yet the current implementation of the Compose is more a POC, than the real implementation that can take full advantage of the Framework and be further evaluated

To have that functionality as first class in the Framework, the above roadmap is still the right past. The first obstacle as noted is fixing how AWS Credentials are handled in a Framework. Unfortunately, it's not a trivial task, but any help is highly appreciated
